### DISCUSSION TITLE
Understanding runtime metrics

### DISCUSSION THREAD

**Discussion by xsoheilalizadeh** [dotnet/runtime#78378] [Q&A]
I'm trying to build a dashboard with runtime the metrics the runtime event counter gives me and I want to understand a few of them to detect issues. 

## ThreadPool
![image](https://user-images.githubusercontent.com/22220356/201903456-fc0398c3-f61d-45e4-b318-c3d34db3326b.png)

Here I have this information to monitor the thread pool,

### How can I understand if I have thread pool starvation here?

If the numbers of **threads** are more than the **completed** it means there is starvation or the number of **queued** is more than **completed**?

## Memory

I see sometimes the ratio goes up 90% but I don't know what's the normal percentage and how I can find the root cause.

![image](https://user-images.githubusercontent.com/22220356/201915777-2cc648a8-da2a-4886-a8d2-7218d7fca9d0.png)

What Time in GC means here if it has a high percentage what are the side effects and how I can find the root cause? 

![image](https://user-images.githubusercontent.com/22220356/201916162-d5565506-cb62-4753-86cd-584ae1930a88.png)

**Comment by xsoheilalizadeh** [comment]
If `ThreadPool.PendingWorkItemCount` is bigger than `ThreadPool.ThreadCount`, it means that there is thread pool starvation? 

@kouvel

**Comment by davidfowl** [comment]
https://learn.microsoft.com/en-us/dotnet/core/diagnostics/debug-threadpool-starvation
